gateway/api/middleware.go

```diff
@@ -51,70 +96,188 @@ func (a *Api) Authenticate(c *gin.Context) {
 		)
 		c.Set(pgusers.ContextLoggerKey, zaplogger.Sugar())
 	}
-	switch openapi.RoleType(roleName) {
-	case openapi.RoleStandardType: // noop
-	case openapi.RoleUnregisteredType:
-		if !ctx.IsEmpty() {
-			break
+
+	c.Set(storagev2.ContextKey,
+		storagev2.NewContext(ctx.UserSubject, ctx.OrgID).
+			WithUserInfo(ctx.UserName, ctx.UserEmail, string(ctx.UserStatus), ctx.UserPicture, ctx.UserGroups).
+			WithSlackID(ctx.UserSlackID).
+			WithOrgName(ctx.OrgName).
+			WithOrgLicense(ctx.OrgLicense).
+			// WithApiURL(a.IDProvider.ApiURL).
+			WithGrpcURL(a.GrpcURL),
+	)
+
+	c.Next()
+}
+
+// AllowApiKey is a middleware that allows the request to proceed if the request has an api key.
+// It sets the "allow-api-key" key in the context to true so the Auth Middleware can authorize it
+func (a *Api) AllowApiKey(c *gin.Context) {
+	c.Set("allow-api-key", true)
+	c.Next()
+}
+
+func (a *Api) apiKeyMiddleware(c *gin.Context) {
+	configuredApiKey := appconfig.Get().ApiKey()
+	apiKeyReq := c.GetHeader("Api-Key")
+	// split apiKey by | getting the first and the second part
+	// the first part is the orgID and the second part is the apiKey
+	orgID := strings.Split(apiKeyReq, "|")[0]
+	// apiKey := strings.Split(apiKeyReq, "|")[1]
+	org, err := pgorgs.New().FetchOrgByID(orgID)
+	if err != nil || org == nil {
+		c.AbortWithStatus(http.StatusUnauthorized)
+		return
+	}
+
+	if apiKeyReq == configuredApiKey {
+		ctx := &pguserauth.Context{
+			OrgID:       orgID, // "400342d2-fa79-4a05-aa8a-b4d4d1989acb",
+			OrgName:     org.Name,
+			OrgLicense:  org.License,
+			UserUUID:    "API_KEY",
```

Comment:
I recommend using a deterministic UUID for this one, I'm not sure how this information is used in the underline layers. We usually rely on UUID type in the database.

To generate a deterministic UUID based on that name:
```go
uuidStr := uuid.NewSHA1(uuid.NameSpaceURL, []byte(`API_KEY`))
```

Reply:
nice one